src/core/qgsproject.h

```diff
@@ -263,18 +263,33 @@ class CORE_EXPORT QgsProject : public QObject, public QgsExpressionContextGenera
      */
     void clear();
 
+    /**
+     * Flags which control project read behavior.
+     * \since QGIS 3.10
+     */
+    enum ReadFlag
+    {
+      FlagDontResolveLayers = 1 << 1, //!< Don't resolve layer paths (i.e. don't load any layer content). Dramatically improves project read time if the actual data from the layers is not required.
```

Comment:
@nyalldawson just a nitpick - `1 << 1` is two, maybe better to start from one? :)

Reply:
Thanks, fixed